test/hotspot/jtreg/compiler/vectorization/runner/LoopArrayIndexComputeTest.java

```diff
@@ -161,16 +161,28 @@ public int[] indexWithTwoInvariants() {
     }
 
     @Test
-    // Note that this case cannot be vectorized due to data dependence.
-    @IR(failOn = {IRNode.STORE_VECTOR})
-    public int[] indexWithDifferentConstants() {
+    // No true dependency in read-forward case.
+    @IR(applyIfCPUFeatureOr = {"asimd", "true", "sse2", "true"},
+        counts = {IRNode.STORE_VECTOR, ">0"})
```

Comment:
You may need add `applyIf = {"AlignVector", "false"}` for these newly added IR check rules.

Reply:
@fg1417 you are right!

I think we should also add `AlignVector` to the IR whitelist.

I also see that in `test/hotspot/jtreg/compiler/vectorization/runner/` a lot of tests have `@requires vm.flagless`. That means we actually do not check any flag combinations with those tests. I think we should file an RFE to make them more general, and add the requirements to the IR rules.

Reply:
Hi @eme64, I'd like to explain more about the `@requires vm.flagless`. Vladimir Kozlov had suggested removing those annotations. I didn't do that before because those annotations cannot be simply removed. All tests under `compiler/vectorization/runner/` are used for both correctness check and vectorizability (IR) check. For correctness check, each test method is invoked twice and the return results from the interpreter and C2 compiled code are compared. We use compiler control via WhiteBox API from the test runner to force these methods running in interpreter and C2 (see the logic in `VectorizationTestRunner.java`). The force compilation would fail if some extra vm option of compiler control (such as `-Xint`) is specified.

A way of removing `@requires vm.flagless` I can think of may be skipping the correctness check in the vectorization test runner if the compiler control fails. I just filed [JDK-8309697](https://bugs.openjdk.org/browse/JDK-8309697) for this. Please let me know if you have any better ideas or suggestions.

Reply:
@pfustc If I run it with `-Xint`, then it says "Test results: no tests selected". I think that is because of `@requires vm.compiler2.enabled`. But sure, there may be some other flags that mess with the compiler controls.

But I think it is important to remove the `@requires vm.flagless`, there are always bugs lurking around with more flag combinations. Plus, we don't have all the hardware that exists out there. That is why it is crucial that we can run with flags `AlignVector` (some ARM machines have it on true) or `UseKNLSetting` (intel), for example.

I'll temporarily add the `@requires vm.flagless` back in for `test/hotspot/jtreg/compiler/vectorization/runner/LoopArrayIndexComputeTest.java`.